proj/src/main/webapp/script.js

```diff
@@ -365,327 +31,6 @@ function appendMessage(payload){
-    })
-}
-
-function initBio() {
-    const bioBox = document.getElementById('user-bio');
-    const editInputBox = document.getElementById('bio-edit');
-
-    bioBox.addEventListener('dblclick', function() {
-        this.hidden = true;
-
-        editInputBox.hidden = false;
-        editInputBox.value = this.innerText
-
-        editInputBox.focus();
-    });
-
-    editInputBox.addEventListener('blur', function() {
-        const uid = firebase.auth().currentUser.uid;
-        const userBioRef = firebase.database().ref('/users/'+uid+'/bio');
-        userBioRef.set(this.value);
-
-        this.hidden = true;
-        bioBox.hidden = false;
-        bioBox.innerText = this.value;
-    });
-}
-
-
-function populateSidebar() {
-    const userTagsRef = firebase.database().ref('/users/'+currentUID+'/allTags');
-    userTagsRef.orderByKey().on('child_added', snap => {
-
-        const chatTag = snap.key;
-        const chatId = snap.val();
-
-        // get last message
-        makePreviewWithLastMessage(chatTag, chatId)
-
-    });
-}
-
-function changeChatOnClick(domElement, tag, chatId) {
-    domElement.addEventListener('click', function() {
-        var dbRefObject = getDbRef(tag, chatId);
-        initRef(dbRefObject);
-    });
-}
-
-function populateProfileSidebar(user) {
-    if (user) {
-    const userRef = firebase.database().ref('/users/'+user.uid)
-    userRef.once('value', snap => {
-        const userObj = {};
-        userObj.photo = user.photoUrl;
-        userObj.uid = user.uid;
-        userObj.fname = snap.val().firstName;
-        userObj.lname = snap.val().lastName;
-        userObj.bio = snap.val().bio;  // there is no bio yet
-        userObj.tags = snap.val().allTags;
-        addUserInfoToDom(userObj)
-        })
-    }
-}
-
-function addUserInfoToDom(userObj) {
-    const profile = document.getElementById('user-profile');
-    profile.querySelector("#user-display-name").innerText = userObj.fname + ' ' + userObj.lname;
-    profile.querySelector("#user-pfp").src = userObj.photo;
-    profile.querySelector("#user-bio").innerText = userObj.bio;
-
-    const tagList = profile.querySelector("#user-tags");
-    for (tag in userObj.tags) {
-        if (userObj.tags.hasOwnProperty(tag)) {
-            const tagNode = document.createElement('li');
-            tagNode.innerText = tag;
-            tagList.appendChild(tagNode);
-        }
-    }
-}
-
-/*
-    Location
- */
-
-var position;
-
-function successCallback(pos){
-    position = pos;
-    console.log(pos);
-}
-
-function errorCallback(err){
-    console.log("error");
-}
-
-function getLocation() {
-    if(navigator.geolocation){
-        navigator.geolocation.getCurrentPosition(successCallback,errorCallback,{timeout:10000, enableHighAccuracy:false});
-    }
-    else{
-        console.log("Error. Geolocation not supported or not enabled");
-    }
-    return;
-}
-
-
 /*
     Chatroom sidebar
  */ 
```

Comment:
Should this section also be part of chat.js?

Reply:
we're trying to separate the files based on the SDK (auth + db/location/notifications which will get a script at some point) and since the chatroom sidebar stuff doesn't need auth + db we're leaving it here